flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/connectors/hive/HiveLookupJoinITCase.java

```diff
@@ -378,6 +378,31 @@ public void testLookupJoinWithLookUpSourceProjectPushDown() throws Exception {
         assertEquals("[+I[1, a], +I[2, b], +I[3, c]]", results.toString());
     }
 
+    @Test
+    public void testLookupJoinPartitionTableWithLookUpSourceProjectPushDown() throws Exception {
```

Comment:
Actually, this failure is not related to PartitionTable. We can just change the test case in `testLookupJoinWithLookUpSourceProjectPushDown` to 
```
TableImpl flinkTable =
                (TableImpl)
                        tableEnv.sqlQuery(
                                "select b.x, b.z from "
                                        + " default_catalog.default_database.probe as p "
                                        + " join bounded_table for system_time as of p.p as b on p.x=b.x");
```
to reproduce it. 
So, I think we can just remove the test `testLookupJoinPartitionTableWithLookUpSourceProjectPushDown` 
and only modify the test case in `testLookupJoinWithLookUpSourceProjectPushDown`.

Reply:
Also, could you please update the title of the pr to be more clear? some thing like
`[FLINK-30679][hive]Fix IndexOutOfBoundsException for Hive lookup join when column pushdown to Hive lookup table source`

Reply:
ok

Reply:
when using project push down，the non-partition table is ok ，for case testLookupJoinWithLookUpSourceProjectPushDown. But the partition table is error